crates/json-abi/src/item.rs

```diff
@@ -373,7 +463,63 @@ impl Error {
     }
 }
 
+impl FromStr for Function {
+    type Err = ParserError;
+
+    #[inline]
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        Self::parse(s)
+    }
+}
+
 impl Function {
+    /// Parses a Solidity function signature string:
+    /// `$name($($inputs),*)$(($($outputs),*))?`
+    ///
+    /// Note that [`state_mutability`](Self::state_mutability) is not parsed
+    /// from the input and is always set to [`StateMutability::Payable`].
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// # use alloy_json_abi::{Function, Param, StateMutability};
+    /// assert_eq!(
+    ///     Function::parse("foo(bool bar)"),
+    ///     Ok(Function {
+    ///         name: "foo".to_string(),
+    ///         inputs: vec![Param::parse("bool bar").unwrap()],
+    ///         outputs: vec![],
+    ///         state_mutability: StateMutability::Payable,
+    ///     }),
+    /// )
+    /// ```
+    ///
+    /// [Function]s also support parsing output tuples:
+    ///
+    /// ```
+    /// # use alloy_json_abi::{Function, Param, StateMutability};
+    /// assert_eq!(
+    ///     Function::parse("toString(uint number)(string s)"),
+    ///     Ok(Function {
+    ///         name: "toString".to_string(),
+    ///         inputs: vec![Param::parse("uint number").unwrap()],
+    ///         outputs: vec![Param::parse("string s").unwrap()],
+    ///         state_mutability: StateMutability::Payable,
```

Comment:
Should we default to payable or non payable? I think non-payable?

Reply:
Payable is the EVM "default", the rest are higher level abstractions

Reply:
non-payable is default

Reply:
In Solidity yes, you have to opt out

Reply:
I guess that makes more sense because we're parsing Solidity strings after all